Dist core p7 20200705

jkeenan opened the pull request:
Various test fixes and $VERSION bumps in files under `dist/`.

Please review.

jkeenan committed:
Dumpvalue.t:  Tighten scope of package variables.

Rename some variables for clarity.

In one place, get() is returning a list, so we should test for contents of
list (via Test::More::is_deeply) rather than for string equality.

jkeenan committed:
IO.t: Suppress one 'redefine' warning

jkeenan committed:
01lib.t:  Adapt for strict-by-default

Get correct scoping for package variable.

Use lexical filehandles.

Correct one spelling error dating back to file's creation in 2001.

jkeenan committed:
Abbrev.t:  Bring this test into the 21st century

'our' instead of 'local'.  No more need for 'local $^W' with
warnings-by-default.

jkeenan committed:
XSLoader.t: Suppress 'redefine' warnings in two places

jkeenan committed:
Abbrev.pm: Increment $VERSION

jkeenan committed:
dist/IO: Increment $VERSION to 1.44 throughout

jkeenan committed:
Increment $VERSION in module-generating script

jkeenan committed:
PathTools: Adapt tests for strict/warnings by default.

No need to declare 'use strict'.

Instead of turning off all warnings within certain blocks, turn off only
those we need and within a tight lexical scope.

Note: This will have to be checked on MacOS and Win32.

jkeenan committed:
Carp: Adapt tests for strict/warnings by default

karenetheridge commented on dist/Text-Abbrev/t/Abbrev.t:
```diff
@@ -7,7 +7,7 @@ use Text::Abbrev;
 
 # old style as reference
-local(%x);
+our %x;
```
Why is this `our` instead of `my`?

jkeenan replied:
Because with `my %x` we get this:
```
$ cd t; ./perl harness -v ../dist/Text-Abbrev/t/Abbrev.t; cd -
Name "main::x" used only once: possible typo at t/Abbrev.t line 13.
Use of uninitialized value $x{"list"} in string eq at t/Abbrev.t line 22.
Use of uninitialized value $x{"a"} in string eq at t/Abbrev.t line 29.

...
```
So I'm leaving it as `our`.

karenetheridge commented on dist/lib/t/01lib.t:
```diff
@@ -27,14 +28,14 @@ BEGIN {
 
-    open(MOD, '>', $Module) || DIE $!;
-    print MOD <<'MODULE';
+    open(my $MOD, '>', $Module) || die $!;
```
`||` should be `or`

karenetheridge commented on dist/lib/t/01lib.t:
```diff
@@ -27,14 +28,14 @@ BEGIN {
 
-    open(MOD, '>', $Module) || DIE $!;
-    print MOD <<'MODULE';
+    open(my $MOD, '>', $Module) || die $!;
```
```suggestion
    open(my $MOD, '>', $Module) or die $!;
```

jkeenan replied:
Done.

atoomic commented on dist/PathTools/t/Spec.t:
```diff
@@ -1,6 +1,5 @@
 #!/usr/bin/perl -w
 
-use strict;
```
we should preserve that `use strict` in dual life modules... the module is going to be installed using older versions of Perl
we could even consider the reverse, by enforcing `use strict; use warnings` on all these tests and files...

jkeenan replied:
Noted, see commit messages in updated p.r.

atoomic commented on dist/PathTools/t/crossplatform.t:
```diff
@@ -1,6 +1,5 @@
 #!/usr/bin/perl -w
 
-use strict;
```
same comment, we should not remove `use strict` for dual life modules

jkeenan replied:
Noted, see commit messages in updated p.r.

atoomic commented on dist/XSLoader/t/XSLoader.t:
```diff
@@ -117,6 +117,7 @@ SKIP: {
 
```
easier to get in upstream by adding a `use warnings` at the top?

jkeenan replied:
Noted, see commit messages in updated p.r.

atoomic commented on dist/Text-Abbrev/t/Abbrev.t:
```diff
@@ -48,7 +48,6 @@ print (($r eq $s)?"ok $test\n":"not ok $test\n"); $test++;
 
 # warnings safe with zero arguments
```
not sure if we should remove the `$^W = 1;` but if we do this comment is dead

jkeenan committed:
Carp_overloadless.t: Restore 'use warnings'

Per discussion with Nicolas R, for distributions in the dual-life directories (dist/
and cpan/) we have to account for the possibility that the distro will be
released separately on CPAN and will need explicit 'use warnings' and/or 'use
strict' statements to continue to work as previously with Perl 5.

In this case, we not only restore 'use warnings' but have to add a 'use
strict' as well so that on Perl 5 the subsequent "no strict 'refs'" makes
sense.  (The "no warnings 'once'" was already commented-out in blead.)

jkeenan committed:
Spec.t: Restore 'use strict'

Per discussion with Nicolas R, for distributions in the dual-life directories (dist/
and cpan/) we have to account for the possibility that the distro will be
released separately on CPAN and will need explicit 'use warnings' and/or 'use
strict' statements to continue to work as previously with Perl 5.

jkeenan committed:
crossplatform.t:  Restore 'use strict'

Per discussion with Nicolas R, for distributions in the dual-life directories (dist/
and cpan/) we have to account for the possibility that the distro will be
released separately on CPAN and will need explicit 'use warnings' and/or 'use
strict' statements to continue to work as previously with Perl 5.

jkeenan committed:
01lib.t: Use lower-precedence 'or' operator

Since we're already changing this line to correct a likely typo, we can
make this correction suggested by ether here as well.

jkeenan commented:
Adopted most of these suggestions.  Letting Dumpvalue.t stay as is for now as it's subject of p.r. to blead.
